### PR TITLE
Add preprocessed_builds to abseil-cpp

### PR DESCRIPTION
**Pull request by veblush**
Added `preprocessed_builds.yaml` to be used as builds for abseil dependency.
Added `preprocessed_builds.yaml.gen.py` to generate `builds.yaml` files from BUILD files of abseil.
There is no change on generated build files yet. 

This is a prerequisite for #20184

### CODE REVIEW

File: src/abseil-cpp/preprocessed_builds.yaml.gen.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      textual_hdrs=normalize_paths(get_elem_value(elem, "textual_hdrs") or []),
+      deps=get_elem_value(elem, "deps") or [],
+      visibility=get_elem_value(elem, "visibility") or [],
+      testonly=get_elem_value(elem, "testonly") or False)
+
+
+def read_build(package):
+  """Runs bazel query on given package file and returns all cc rules."""
+  result = subprocess.check_output(
+      ["bazel", "query", package + ":all", "--output", "xml"])
```
Comment (gnossen) [inline src/abseil-cpp/preprocessed_builds.yaml.gen.py]: Nice approach! :D

File: src/abseil-cpp/preprocessed_builds.yaml.gen.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python2.7
```
Comment (gnossen) [inline src/abseil-cpp/preprocessed_builds.yaml.gen.py]: Let's not force new code to a version of Python that will be officially EOL in less than a month. Either `#!/usr/bin/env python3` or no shebang at all.

File: src/abseil-cpp/preprocessed_builds.yaml.gen.py (hunk 1 above)
Comment (veblush) [reply src/abseil-cpp/preprocessed_builds.yaml.gen.py]: :)

**Comment by veblush** [comment]
If someone forget running the script when upgrading abseil, it results in build failure related with abseil.

File: src/abseil-cpp/preprocessed_builds.yaml.gen.py (hunk 2 above)
Comment (veblush) [reply src/abseil-cpp/preprocessed_builds.yaml.gen.py]: Changed it to have python3 shebang.

**Pull request merged**